r/R/filesystem.R

```diff
@@ -547,6 +561,11 @@ GcsFileSystem$create <- function(anonymous = FALSE, retry_limit_seconds = 15, ..
     )
   }
 
+  # Stop if expiration isn't a POSIXct
+  if ("expiration" %in% names(options) && !inherits(options$expiration, "POSIXct")) {
+    stop("Option 'expiration' must be of class POSIXct", call. = FALSE)
```

Comment:
You could extend this to make it even more helpful by printing out the class of the object, e.g. something like

`paste("Option 'expiration' must be of class POSIXct, not", class(options$expiration)[[1]])`

Only a minor point - feel free to leave it as-is.

Reply:
Thanks, this is an improvement.